utilities/hgn_result.py

```diff
@@ -55,22 +65,18 @@ def append_reconstruction(self, reconstruction):
         """Append guessed reconstruction
 
         Args:
-            reconstruction (torch.Tensor): Decoder of the HGN output
+            reconstruction (torch.Tensor): Tensor of shape (seq_len, channels, height, width)
+                containing the reconstructed rollout.
         """
-        if self.reconstructed_rollout is None:
-            self.reconstructed_rollout = reconstruction
-        else:
-            self.reconstructed_rollout = torch.cat(
-                (self.reconstructed_rollout, reconstruction), dim=1)
+        assert self.reconstruction_ptr < self.reconstructed_rollout.shape[1],\
```

Comment:
I dont understand why do you limit this. What if I wanna do rollouts longer than the inputed sequence to analyse how well the system continues to evolve after the inputed data? Same with going backward in time before the provided rollout.

Reply:
The first reason I changed the code is that using `torch.cat()` every time we append means that we are copying the whole tensor at every call.

`self.reconstructed_rollout` is initially instantiated with the shape of the batch, so it will always work as long as you use the same length of your ground truth data. I haven't thought about the issue of testing with a length higher than your ground truth data. But wouldn't be better, in that case, to just take a longer sequence as ground truth? Otherwise we cannot compare it.